src/OrchardCore/OrchardCore.Search.AzureAI.Core/Services/AzureAISearchIndexManager.cs

```diff
@@ -165,21 +174,22 @@ private string GetIndexPrefix()
     {
         if (!_memoryCache.TryGetValue<string>(_prefixCacheKey, out var value))
         {
-            var builder = new StringBuilder();
+            var prefix = _shellSettings.Name.ToLowerInvariant();
 
             if (!string.IsNullOrWhiteSpace(_azureAIOptions.IndexesPrefix))
             {
-                builder.Append(_azureAIOptions.IndexesPrefix.ToLowerInvariant());
-                builder.Append('-');
+                prefix = $"{_azureAIOptions.IndexesPrefix.ToLowerInvariant()}-{prefix}";
```

Comment:
I don't really see how these changes fix the issue (or where that issue is). The prefix will now be `<shell name>-<configured prefix>` instead of `<configured prefix>-<shell name>`, with just `<shell name>` if no prefix is configured. Can you help me understand?

Reply:
this line does not fix the problem but improve the performance since we are not using a large string, concatenation is cheaper than using StringBuilder. the fix is is line 46 (I tagged you there)